File: libwebrtc-sys/src/rtc_buffer.cc
#include "libwebrtc-sys/include/rtc_buffer.h"

BufferUint8::BufferUint8(rtc::Buffer* buffer): buffer(buffer) {}

rtc::Buffer* BufferUint8::GetBuffer() const {
    return buffer;
}

void BufferUint8::append_data(rust::Slice<const uint8_t> data) {
    buffer->AppendData(data.data(), data.size());
}

File: libwebrtc-sys/src/rtc_buffer.rs
use cxx::UniquePtr;

#[cxx::bridge]
pub mod ffi {
    unsafe extern "C++" {
        include!("include/rtc_buffer.h");
        type BufferUint8;

        fn append_data(self: Pin<&mut BufferUint8>, data: &[u8]);
        fn gen_unique_rtc_buffer_u8() -> UniquePtr<BufferUint8>;
    }
}